ui_testing/pages/order_page.py

```diff
@@ -179,13 +180,59 @@ def get_last_order_row(self):
     def get_shipment_date(self):
         return self.base_selenium.get_value(element='order:shipment_date')
 
-    def get_test_date(self):
-        return self.base_selenium.get_value(element='order:test_date')
+    def get_random_suborder_data(self, row_id=None):
+        
+        # get all the suborders
+        suborders = self.base_selenium.get_table_rows(element='order:suborder_table')
+        # select random row id
+        if not row_id:
+            if len(suborders) > 1:
+                row_id = randint(0, len(suborders) - 1)
+            else:
+                row_id = 0
+
+        self.info('Getting suborder no. {}'.format(row_id))
+
+        # get the current suborder data
+        suborder = self.base_selenium.get_row_cells_id_dict_related_to_header(
+            row=suborders[row_id], table_element='order:suborder_table')
+        # attach the row_id to the suborder
+        suborder['row_id'] = row_id
+        # attach the row element to the suborder
+        suborder['row_element'] = suborders[row_id]
+        return suborder
+
+    def open_suborder_edit_mode(self, row_id=None):
+        # get the suborder
+        suborder_data = self.get_random_suborder_data(row_id)
+        self.info('Open suborder no. {} for edit mode'.format(suborder_data['row_id']))
+        # click the table row (used script because the tr is not clickable)
+        self.base_selenium.driver.execute_script('arguments[0].click();', suborder_data['row_element'])
+        return suborder_data['row_id']
+
+    def close_suborder_edit_mode(self):
+        self.info('Close the suborder edit mode')
+        webdriver.ActionChains(self.base_selenium.driver).send_keys(Keys.ESCAPE).perform()
+
```

Comment:
you can use update_suborder with few changes to do your magic,
will just send the index of the order, select table records, 
records[index].click()
will open this row in edit mode.
and in the parameters you can send whatever attribute you desire to be updated